### ISSUE TITLE
Cached request feature throwing an error

### ISSUE THREAD

**Issue by audas** [RamParameswaran/pyAFL#9]
When making multiple requests to different URLS (requesting multiple player data for example) after 4 or 5 requests it throws an error from players\models.py

`
**return request('get', url, params=params, **kwargs)

File "C:\Users\MyName\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
    and self._filter_fn(response) is not True
AttributeError: _ConfiguredCachedSession object has no attribute _filter_fn**`

**Comment by RamParameswaran** [comment]
Thanks again @audas. This one is interesting.
Would you mind sending the code that you ran that raised this error?

This bug clearly relates to [`pyAFL/requests/requests.py` ln 56](https://github.com/RamParameswaran/pyAFL/blob/015db0d4c68346585b46c6a47df071d701dce774/pyAFL/requests/requests.py#L56). This particular bit of code inherits and overwrites the `CachedSession` class from the ["requests_cache"](https://github.com/reclosedev/requests-cache) library. Incidentally, the "requests_cache" library released a number of minor and major changes in April 2021, so it may be worth bumping the "request_cache" dependency to the latest version and updating the pyAFL code accordingly. I'll look into this.

**Comment by audas** [comment]
I couldn't get force_live to work so I hard coded it. I then forced the config url onto the returned player url and this fixed it.

I am trying to get ALL the player ages - I think debut would be good too. 

So in the return value of def _get_player_url() I placed a hack - 

return config.AFLTABLES_STATS_BASE_URL +"stats/"+url_list[0].attrs.get("href")

**Comment by JWCook** [comment]
In general, modifying `CachedSession.request()` is likely to break with every requests-cache release, since the internals of that method are going to be constantly changing. The signature will stay the same, though, so overriding/wrapping it is a good option.

If you want to modify cached response content, you can call `CachedSession.request()`, modify the response object, and then save it again:

```python
class AFLTablesCachedSession(CachedSession):
    def request(self, method, url, **kwargs):
        response = super().request(method, url, **kwargs)
        self.cache.save_response(self.post_process(response))

    def post_process(response):
        # Do whatever you want to the response object here
        return response
```

Support for that pattern was added in [0.6.0](https://github.com/reclosedev/requests-cache/blob/master/HISTORY.md#060-2021-04-09). Hope that helps!

**Comment by RamParameswaran** [comment]
Thanks @JWCook for the great suggestion. :v: 
I'll create a patch which follows this pattern shortly.

**Comment by JWCook** [comment]
Whoops, I just realized I gave you the wrong info. I thought that feature made it into 0.6, but it's actually in 0.7, which is still in development. If you want, you can try out the latest pre-release build with:
```
pip install -U --pre requests-cache
```

Note that pre-releases are more likely to contain bugs, but you're welcome to [make issues](https://github.com/reclosedev/requests-cache/issues/new/choose) on the requests-cache repo for bug reports or feature requests. Otherwise I could probably come up with a different example that works with 0.6.

**Comment by RamParameswaran** [comment]
Thanks @JWCook. Do you know the expected release date for 0.7? The read-the-docs [suggests it is this month](https://requests-cache.readthedocs.io/en/latest/history.html)...

I really like the pattern you suggested - and so if it's a matter of waiting a few weeks for 0.7 to release, then I think it's worth the wait.

**Comment by JWCook** [comment]
Yes, I expect the next release will be by the end of the month.

**Comment by RamParameswaran** [comment]
Closed by #20
